utils/utils.py

```
@@ -254,4 +267,48 @@ async def process_summary_api_response(context, update, edit_response, processin
+    return True
+
+
+def extract_keywords(message):
+    """Extract meaningful keywords by removing stop words and short words."""
+    stop_words = {'the', 'a', 'an', 'and', 'or', 'but', 'is', 'are', 'was', 'were', 'be', 'been', 'being',
+                   'have', 'has', 'had', 'do', 'does', 'did', 'will', 'would', 'shall', 'should', 'may',
+                   'might', 'must', 'can', 'could', 'i', 'you', 'he', 'she', 'it', 'we', 'they', 'my',
+                   'your', 'his', 'her', 'its', 'our', 'their', 'this', 'that', 'these', 'those', 'what',
+                   'which', 'who', 'whom', 'whose', 'where', 'when', 'why', 'how', 'in', 'on', 'at', 'by',
+                   'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before',
+                   'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over',
```

Ruff (0.8.2)

Sets should not contain duplicate item `'in'`

Remove duplicate item

(B033)